### PR TITLE
[TF FE] Add Conv layer tests

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Conv2D layer test
 - Conv3D layer test

### Tickets:
 - 92761, 92763

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # input_filter - should be an array, defines a filter
+    # input_strides - should be an array, defines strides of a sliding window to use
+    # input_padding - should be a string, defines padding algorithm
+    # ir_version - common parameter
+    # use_new_frontend - common parameter
+    def create_conv2d_placeholder_const_net(self, input_shape, input_filter, input_strides, input_padding, dilations, ir_version, use_new_frontend):
+        """
+            Tensorflow net                  IR net
+
+            Placeholder->Conv2D    =>       Placeholder->Transpose->Convolution--------------------->Reshape
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py:25]: nit: Placeholder->Transpose->Convolution->Transpose?
Please correct this in the next PR

File: tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    test_data = [
+        dict(input_shape=[1, 10, 10, 1], input_filter=[3, 3, 1, 1], input_strides=[1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 2], input_filter=[3, 3, 2, 2], input_strides=[1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 2], input_filter=[3, 3, 2, 1], input_strides=[1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 3], input_filter=[2, 2, 3, 3], input_strides=[1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 16, 16, 3], input_filter=[2, 2, 3, 3], input_strides=[1, 2, 2, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 4], input_filter=[2, 2, 4, 2], input_strides=[1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 16, 16, 3], input_filter=[2, 2, 3, 3], input_strides=[1, 2, 2, 1], dilations=[1, 2, 2, 1]),
+        pytest.param(
+            dict(input_shape=[1, 224, 224, 3], input_filter=[16, 16, 3, 1], input_strides=[1, 2, 2, 1], dilations=[1, 2, 2, 1]),
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py:70]: `input_filter=[16, 16, 3, 1]` is really huge kernel, not practical, and may consume solid testing time. Please replace with the other filter as `input_filter=[4, 4, 3, 2]`.

File: tests/layer_tests/tensorflow_tests/test_tf_Conv3D.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        dict(input_shape=[1, 10, 10, 10, 1], input_filter=[3, 3, 3, 1, 1], input_strides=[1, 1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 10, 2], input_filter=[3, 3, 3, 2, 2], input_strides=[1, 1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 10, 2], input_filter=[3, 3, 3, 2, 1], input_strides=[1, 1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 16, 16, 16, 3], input_filter=[4, 2, 4, 3, 3], input_strides=[1, 2, 2, 2, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 20, 3], input_filter=[2, 4, 2, 3, 3], input_strides=[1, 2, 2, 2, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 10, 4], input_filter=[3, 3, 3, 4, 2], input_strides=[1, 1, 1, 1, 1], dilations=None),
+        dict(input_shape=[1, 10, 10, 20, 3], input_filter=[2, 4, 2, 3, 3], input_strides=[1, 2, 2, 2, 1], dilations=[1, 2, 2, 2, 1]),
+        pytest.param(
+        dict(input_shape=[1, 224, 224, 224, 3], input_filter=[16, 10, 13, 3, 3], input_strides=[1, 2, 2, 2, 1], dilations=None),
+            marks=pytest.mark.precommit_tf_fe),
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Conv3D.py:62]: the same comment. Please replace with `input_filter=[1, 2, 3, 3, 2]` as example

**Review by rkazants — COMMENTED** [review]
Left some comments. Please address.

File: tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py (hunk 1 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py]: Fixed. But I'm sure I saw such workflow in IR's... Never mind...

File: tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py (hunk 2 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_Conv2D.py]: done

File: tests/layer_tests/tensorflow_tests/test_tf_Conv3D.py (hunk 3 above)
Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_Conv3D.py]: done

**Review by rkazants — APPROVED** [review]

**Pull request merged**